### ISSUE TITLE
[enhancement]: PublishPipelineArtifact/DownloadPipelineArtifact for On Prem / Server

### ISSUE THREAD

**Issue by msherms2** [microsoft/azure-pipelines-tasks#19698]
### Task name

PublishPipelineArtifact/DownloadPipelineArtifact

### Describe your feature request here

I cannot for the life of me find any reason that this is not available for On Prem / Server. Both equivalent PublishBuildArtifact/DownloadBuildArtifact operations are horribly inefficient and surely can be improved. Given you're able to use it with services and thus over a connection, surely this should easily be supported for On prem.

If you can't just enable them for On Prem, can you improve the disastrous performance of the Build equivalents? For many small files it falls apart, and ultimately seems like an unnecessary problem to have in 2024.

**Comment by msherms2** [comment]
It isn't stale, they're just ignoring it.

**Comment by msherms2** [comment]
We just updated over the weekend so I only have the old agent version (2.210.1). We host our own backing VMs or docker containers to use in the pipelines. 

https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-pipeline-artifact-v1?view=azure-pipelines

This straight out says it is not supported, but no reasoning why. Any ideas? I did confirm that if i use it, the error it states does occur.

**Comment by msherms2** [comment]
Yeah, I was all set to try it and then realized your link above is for services, as well. As soon as you change it to server 2022 it tells you version 0 is also not supported.

I guess I get not spending effort if they plan to deprecate server, which probably isn't a very good decision given that businesses like DOD stuff exist that can't use cloud. But either way, yeah, fixing DownloadBuildArtifact and PublishBuildArtifact from being so lousy performance wise would be fine, too. I'm hoping that this gets traction but I'm suspecting it won't be investigated. Thank you for at least looking!

**Comment by msherms2** [comment]
Still not stale. Still just being ignored.

**Comment by ToreDemant** [comment]
We are also hit by this as we have repositories both on ADO Services and ADO Server (latest, 2022 Update 2) and share some pipeline templates between pipelines on both sides.

One problem is the trouble of having to use different tasks when on services/on-prem, but it also seems that new development and optimizations are only happening on PublishPipelineArtifact/DownloadPipelineArtifact, leaving the others in an antiquated state:
https://github.com/microsoft/azure-pipelines-tasks/issues/6830 / https://github.com/microsoft/azure-pipelines-agent/issues/1480
https://github.com/microsoft/azure-pipelines-tasks/issues/19829

**Comment by ToreDemant** [comment]
The block was added six years ago to PipelineArtifactPluginV1/PipelineArtifactPluginV2, but the pull request does not contain a reason for it being blocked:
https://github.com/microsoft/azure-pipelines-agent/pull/2425

**Comment by ToreDemant** [comment]
Also asked the question here: https://devblogs.microsoft.com/devops/announcing-the-new-azure-devops-server-rc-release/?commentid=5091#comment-5091

**Comment by ToreDemant** [comment]
@tarunramsinghani I am sorry that I keep bugging you, but any chance that you know why or know who can tell us why this have not been fixed?

**Comment by ToreDemant** [comment]
> The block was added six years ago to PipelineArtifactPluginV1/PipelineArtifactPluginV2, but the pull request does not contain a reason for it being blocked: [microsoft/azure-pipelines-agent#2425](https://github.com/microsoft/azure-pipelines-agent/pull/2425)

@jahsu-MSFT / @fadnavistanmay can you help us understand this?

**Comment by ToreDemant** [comment]
Still not stale. Still just being ignored.
